### PR TITLE
add snapshot support for subnetworks

### PR DESCRIPTION
**Pull request by evanmcc**
like it says on the tin.

### CODE REVIEW

**Review by andymck — APPROVED** [review]
lgtm, snapshots are still somewhat vague to me tho

File: src/ledger/v1/blockchain_ledger_v1.erl [hunk 1]
```diff
@@ -6205,6 +6219,33 @@ load_h3dex(H3DexList, Ledger) ->
             ok
     end.
 
+-spec snapshot_subnetworks(ledger()) -> [{binary(), binary()}].
+snapshot_subnetworks(Ledger) ->
```
Comment (Vagabond) [inline src/ledger/v1/blockchain_ledger_v1.erl:6223]: Should this be snapshot_raw_subnetworks?

Comment (Vagabond) [reply src/ledger/v1/blockchain_ledger_v1.erl:6223]: No, never mind

**Review by Vagabond — APPROVED** [review]

File: src/ledger/v1/blockchain_ledger_snapshot_v1.erl [hunk 2]
```diff
@@ -339,7 +342,9 @@ generate_snapshot_v5(Ledger0, Blocks, Infos, Mode) ->
                 {upgrades         , Upgrades},
                 {net_overage      , NetOverage},
                 {hnt_burned       , HntBurned},
-                {validator_count  , ValidatorCount}
+                {validator_count  , ValidatorCount},
+                {subnetworks      , Subnetworks},
+                {accounts_v2      , AccountsV2}
```
Comment (xandkar) [inline src/ledger/v1/blockchain_ledger_snapshot_v1.erl:347]: Need to also add these to the key type(s). Annoying that Dialyzer doesn't catch this.

**Review by vihu — APPROVED** [review]
Looks good, may be good to add a simple test for this.

File: src/ledger/v1/blockchain_ledger_snapshot_v1.erl (hunk 2 above)
Comment (vihu) [reply src/ledger/v1/blockchain_ledger_snapshot_v1.erl:349]: @xandkar can you plz review [805c321](https://github.com/helium/blockchain-core/pull/1415/commits/805c321e95bdf734c845efd8c92b71eb1ba74218) and does it need to go anywhere else?

Comment (xandkar) [reply src/ledger/v1/blockchain_ledger_snapshot_v1.erl:349]: @vihu `accounts_v2` should go into `key_raw()`

**Review by xandkar — APPROVED** [review]

**Pull request merged**